crates/biome_service/src/configuration.rs

```diff
@@ -295,10 +295,11 @@ pub fn load_editorconfig(
     fs: &dyn FileSystem,
     workspace_root: Utf8PathBuf,
 ) -> Result<(Option<Configuration>, Vec<EditorConfigDiagnostic>), WorkspaceError> {
-    // How .editorconfig is supposed to be resolved: https://editorconfig.org/#file-location
-    // We currently don't support the `root` property, so we just search for the file like we do for biome.json
-    if let Some(auto_search_result) = fs.auto_search_file(&workspace_root, ".editorconfig") {
-        let AutoSearchResult { content, .. } = auto_search_result;
+    let file_path = workspace_root.join(".editorconfig");
+
+    // We only need to search `.editorconfig` in the `workspace_root` the same as the `biome.json`,
+    // because now `.editorconfig` is enable by default, so we don't need to search upwards to avoid conflict with `biome.json`.
```

Comment:
That's not correct. `biome.json` can come from a folder that isn't the project root, but maybe a parent folder; that's how we load the configuration.

So the question is: do we pass the workspace directory or the directory where `biome.json` was found?

Reply:
i think workspace directory is more reasonable, how about @arendjr ?

Reply:
Looking at the call sites where `load_editorconfig()` is called, it appears it's called with the directory where `biome.json` was found, with a fallback for the current working directory if no `biome.json` was found.

To me this looks *mostly* correct, but the one thing it is missing is that we may want to search upwards if **no** `biome.json` was found. Imagine someone working on personal projects, without creating explicit `biome.json` files in each, but they do have an `.editorconfig` in their home directory. They'd like the `.editorconfig` to be picked up.

Reply:
i update the case here, i use the workspace directory to search `.editorconfig` here, if find it, i will compare the path with `biome.json` path, if it's a parent path for `biome.json`, i will ignore it or it will reuse.